packages/itwin/reports-config-widget/src/widget/components/Reports.tsx

```diff
@@ -125,140 +106,123 @@ export const Reports = () => {
-                      )}
-                    </Button>
-                  </div>
-                </>
-              </EmptyMessage>
-            ) : (
-              <div className="rcw-reports-list">
-                {filteredReports.map((report) => (
-                  <ReportHorizontalTile
-                    key={report.id}
-                    report={report}
-                    onClickTitle={() => {
-                      setSelectedReport(report);
-                      setReportsView(ReportsView.REPORTSMAPPING);
-                    }}
-                    jobStartEvent={jobStartEvent}
-                    bulkExtractor={bulkExtractor}
-                    onClickDelete={() => {
-                      setSelectedReport(report);
-                      setShowDeleteModal(true);
-                    }}
-                    onClickModify={() => {
-                      setSelectedReport(report);
-                      setReportsView(ReportsView.MODIFYING);
-                    }}
-                    onSelectionChange={onSelectionChange}
-                    selected={selectedReportIds.some((reportId) => report.id === reportId)}
-                  />
-                ))}
+                </Button>
               </div>
-            )}
-          </Surface>
-          <DeleteModal
-            entityName={selectedReport?.displayName ?? ""}
-            show={showDeleteModal}
-            setShow={setShowDeleteModal}
-            onDelete={async () => {
-              const reportsClientApi = new ReportsClient(
-                generateUrl(REPORTING_BASE_PATH, apiConfig.baseUrl)
-              );
-              const accessToken = await apiConfig.getAccessToken();
-              await reportsClientApi.deleteReport(
-                accessToken,
-                selectedReport?.id ?? ""
-              );
-            }}
-            refresh={refresh}
-          />
-        </>
-      );
-  }
+            </>
+          </EmptyMessage>
+        ) : (
+          <div className="rcw-reports-list">
+            {filteredReports.map((report) => (
+              <ReportHorizontalTile
+                key={report.id}
+                report={report}
+                onClickTitle={onClickReportTitle}
+                jobStartEvent={jobStartEvent}
+                bulkExtractor={bulkExtractor}
+                onClickDelete={() => {
+                  setShowDeleteModal(report);
+                }}
+                onClickModify={onClickReportModify}
+                onSelectionChange={onSelectionChange}
+                selected={selectedReportIds.some(
+                  (reportId) => report.id === reportId
+                )}
+              />
+            ))}
+          </div>
+        )}
+      </Surface>
+      <DeleteModal
+        entityName={showDeleteModal?.displayName}
+        onClose={() => setShowDeleteModal(undefined)}
+        onDelete={async () => {
+          const reportsClientApi = new ReportsClient(
```

Comment:
I see that ReportsClient is getting created in multiple places. Maybe it could be handled in a context somewhere? That way you will not need to pass around the baseUrl.

Reply:
I move all the client creation to the context. baseUrl is still needed in one location. Which is to generate the `odataFeedUrl`.